redisinsight/ui/src/pages/rdi/connection-form/ConnectionForm.tsx

```diff
@@ -0,0 +1,213 @@
+import {
+  EuiButton,
+  EuiFieldPassword,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiIcon,
+  EuiSpacer,
+  EuiSplitPanel,
+  EuiTitle,
+  EuiToolTip
+} from '@elastic/eui'
+import { Field, FieldInputProps, Form, Formik, FormikErrors } from 'formik'
+import React, { useEffect, useState } from 'react'
+
+import { RdiInstance } from 'uiSrc/slices/interfaces'
+import ValidationTooltip from './components/ValidationTooltip'
+
+import styles from './styles.module.scss'
+
+export interface ConnectionFormValues {
+  name: string
+  url: string
+  username: string
+  password: string
+}
+
+export interface Props {
+  onAddInstance: (instance: Partial<RdiInstance>) => void
+  onCancel: () => void
+  editInstance: RdiInstance | null
+  isLoading: boolean
+}
+
+const buildInitialValues = (values: RdiInstance | null): ConnectionFormValues => ({
+  name: values?.name || '',
+  url: values?.url || '',
+  username: values?.username || '',
+  password: values?.password || ''
+})
+
+const UrlTooltip = () => (
+  <EuiToolTip
+    data-testid="connection-form-url-tooltip"
+    title={(
+      <div>
+        <p>
+          <b>Pasting a connection URL auto fills the instance details.</b>
+        </p>
+        <p style={{ margin: 0, paddingTop: '10px' }}>The following connection URLs are supported:</p>
+      </div>
+    )}
+    className={styles.urlTooltip}
+    anchorClassName="inputAppendIcon"
+    position="right"
+    content={(
+      <ul>
+        <li>
+          <span />
+          TBD
+        </li>
+      </ul>
+    )}
+  >
+    <EuiIcon data-testid="connection-form-url-icon" type="iInCircle" style={{ cursor: 'pointer' }} />
+  </EuiToolTip>
+)
+
+const ConnectionForm = ({ onAddInstance, onCancel, editInstance, isLoading }: Props) => {
+  const [initialFormValues, setInitialFormValues] = useState(buildInitialValues(editInstance))
+
+  useEffect(() => {
+    setInitialFormValues(buildInitialValues(editInstance))
+  }, [editInstance])
+
+  const onSubmit = (formValues: ConnectionFormValues) => {
+    onAddInstance(formValues)
+  }
+
+  const validate = (values: ConnectionFormValues) => {
+    const errors: FormikErrors<ConnectionFormValues> = {}
+
+    if (!values.url) {
```

Comment:
May be will be better to create 
`const fieldDisplayNames = { url: 'URL', username: 'Username, password: 'Password'`
and use 
`    Object.entries(values).forEach(
      ([key, value]) => !value && Object.assign(errs, { [key]: fieldDisplayNames[key] })
    )`
    As we use in other places.

Reply:
Fair suggestion. Made the change.

Reply:
I tried implementing this but hit an issue where a field in the form `values` wasn't included in the `fieldDisplayNames`. `name` was getting set to `undefined` which was causing some issues. Fwiw, I followed the Formik docs for validation: https://formik.org/docs/guides/validation

If you feel strongly about changing the implementation, let me know.

Reply:
No worry, i'm okay with it. :)